runtime/src/bank.rs

```diff
@@ -795,12 +795,10 @@ impl Bank {
         // TODO: put this assert back in
         // assert!(!self.is_frozen());
         inc_new_counter_debug!("bank-register_tick-registered", 1);
+        let mut w_blockhash_queue = self.blockhash_queue.write().unwrap();
         let current_tick_height = self.tick_height.fetch_add(1, Ordering::Relaxed) as u64;
```

Comment:
How about moving tick_height into a mutex instead of the atomic, so that we can hold the write lock until after blockhash_queue is updated?

Reply:
I rather like the current abstraction of `tick_height` as a measure of time without explicit synchronization, makes it less likely we introduce contention unwittingly.

Also in this specific section I think taking the write lock on `blockhash_queue` communicates rather clearly that we are entering a critical section for the Blockhash, maybe what would help further clarify this is explaining the need in a comment?

Alternatively, we could update the blockhash queue before we update the tick height?

Reply:
I see how this unblocks the dry run. But did that longer blockhash_queue lock affect TPS?  The tick_height mutex suggestion was intended as an approach to avoid holding that blockhash_queue lock.